server/lib/src/credential/apppwd.rs

```diff
@@ -0,0 +1,37 @@
+use crate::credential::{CryptoPolicy, Password};
+use crate::prelude::*;
+use kanidm_proto::v1::OperationError;
+
+#[derive(Debug, Clone)]
+pub struct ApplicationPassword {
+    pub uuid: Uuid,
+    pub(crate) application: Uuid,
+    pub(crate) label: String,
+    pub(crate) password: Password,
+}
+
+impl ApplicationPassword {
+    pub fn new(
+        application: Uuid,
+        label: &str,
+        cleartext: &str,
```

Comment:
crypto policy should be an argument to this function. Cryptopolicy minimum is NOT what you want because when we start up  we do a calculation to find a baseline that's required on this CPU (https://github.com/kanidm/kanidm/blob/master/server/lib/src/idm/server.rs#L151) . We store this in IDM Server Write transactions for you to use, https://github.com/kanidm/kanidm/blob/master/server/lib/src/idm/server.rs#L132 . This way you can pass it in to password hash calls such as here https://github.com/kanidm/kanidm/blob/master/server/lib/src/idm/server.rs#L1707